src/Modal/Modal.jsx

```diff
@@ -29,7 +31,30 @@ class Modal extends Component {
   }
 
   render() {
-    const { onClose, role, ariaLabel, children } = this.props;
+    const {
+      onClose,
+      onCancel,
+      onSubmit,
+      role,
+      ariaLabel,
+      children,
+      submitText,
+      cancelText
+    } = this.props;
+
+    if (onCancel && !cancelText) {
```

Comment:
This kind of stuff should be done as a custom prop type: https://stackoverflow.com/questions/42299335/react-props-set-isrequired-on-a-prop-if-another-prop-is-null-empty